apps/studio/src/lib/db/clients/libsql.ts

```diff
@@ -0,0 +1,155 @@
+import _ from "lodash";
+import rawLog from "electron-log";
+import { SqliteClient, SqliteResult } from "./sqlite";
+import Client_Libsql from "@libsql/knex-libsql";
+import { BasicDatabaseClient } from "./BasicDatabaseClient";
+import Database from "libsql";
+import { LibSQLCursor, LibSQLCursorOptions } from "./libsql/LibSQLCursor";
+import { IDbConnectionDatabase, IDbConnectionServer } from "../types";
+import { SqliteCursor } from "./sqlite/SqliteCursor";
+import { createSQLiteKnex } from "./sqlite/utils";
+
+const log = rawLog.scope("libsql");
+const knex = createSQLiteKnex(Client_Libsql);
+
+/**
+ * FIXME: This class doesn't support returning query data as arrays so
+ * "select 1 as a, 2 as a" will be returned as [{a:2}]. Two ways we can resolve
+ * this:
+ * 1. Fix this in libsql-js https://github.com/tursodatabase/libsql-js/issues/116
+ * 2. Use @libsql/client instead of libsql-js, but this seems to require us
+ *    to use node >= 18
+ */
+export class LibSQLClient extends SqliteClient {
```

Comment:
Yes! If it's not the array mode bug, it would be much smaller! It's a different story if we use [@libsql/client](https://github.com/tursodatabase/libsql-client-ts) since that one isn't compatible with `better-sqlite3`.